src/ess.jl

```diff
@@ -311,43 +341,18 @@ function ess_rhat(
     return ess, rhat
 end
 
-"""
-	copyto_split!(out::AbstractMatrix, x::AbstractMatrix)
-
-Copy the elements of matrix `x` to matrix `out`, in which each column of `x` is split.
+function ess_rhat_bulk(x::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
+    return ess_rhat(Statistics.mean, rank_normalize(x); kwargs...)
+end
 
-If the number of rows in `x` is odd, the sample at index `(size(x, 1) + 1) / 2` is dropped.
-"""
-function copyto_split!(out::AbstractMatrix, x::AbstractMatrix)
-    # check dimensions
-    nrows_out, ncols_out = size(out)
-    nrows_x, ncols_x = size(x)
-    ncols_out == 2 * ncols_x || throw(
-        DimensionMismatch(
-            "the output matrix must have twice as many columns as the input matrix"
-        ),
-    )
-    nrows_out == nrows_x ÷ 2 || throw(
-        DimensionMismatch(
-            "the output matrix must have half as many rows as as the input matrix"
-        ),
+function ess_tail(
+    x::AbstractArray{<:Union{Missing,Real},3}; tail_prob::Real=1//10, kwargs...
+)
+    T = Base.promote_eltype(x, tail_prob)
```

Comment:
What's the reason for the promotion? Shouldn't it work fine without promotion?
```suggestion
```

Reply:
This is a workaround for https://github.com/JuliaStats/Statistics.jl/issues/136.